mypyc/test-data/irbuild-constant-fold.test

```diff
@@ -233,20 +226,241 @@ L0:
     a = 12
     return 1
 
+[case testFloatConstantFolding]
+def bin_ops() -> None:
+    add = 0.5 + 0.5
+    add_mul = (1.5 + 3.5) * 5.0
+    sub = 7.0 - 7.5
+    div = 3.0 / 2.0
+    floor_div = 3.0 // 2.0
+def unary_ops() -> None:
+    neg1 = -5.5
+    neg2 = --1.5
+    neg3 = -0.0
+    pos = +5.5
+def pow() -> None:
+    p0 = 16.0**0
+    p1 = 16.0**0.5
+    p2 = (-5.0)**3
+    p3 = 0.0**0.0
+def error_cases() -> None:
+    div = 2.0 / 0.0
+    floor_div = 2.0 // 0.0
+    power_imag = (-2.0)**0.5
+    power_overflow = 2.0**10000.0
```

Comment:
While floats usually don't overflow and raise an error, this is the one exception I know of. This is why in `mypy/constant_fold.py` there's a guard against OverflowError.